src/type.cpp

```diff
@@ -158,6 +158,17 @@ std::string StructType::id() const noexcept {
   return id_;
 }
 
+std::size_t StructType::offset(const std::string& id) const noexcept {
+  for (auto i = std::size_t{0}, e = fields_.size(); i < e; ++i) {
+    const auto& field = fields_.at(i);
+    if (field->id == id) {
+      return i * field->type->size();
+    }
+  }
+
+  return -1;
```

Comment:
1. `std::size_t` is unsigned, so we cannot return -1. If we need to return a special value when the `id` is not a record member, please mention this in the function comment. Alternatively, we can throw an exception here, as the type-checking phase should have already verified whether the `id` is a record member. Missing this check here would indicate an internal error.  
2. Is the size of each field always the same? If not, we cannot directly calculate the offset by multiplying by their index. Instead, we need to add the sizes of the fields one by one.

Reply:
> 1. `std::size_t` is unsigned, so we cannot return -1. If we need to return a special value when the `id` is not a record member, please mention this in the function comment. Alternatively, we can throw an exception here, as the type-checking phase should have already verified whether the `id` is a record member. Missing this check here would indicate an internal error.
> 2. Is the size of each field always the same? If not, we cannot directly calculate the offset by multiplying by their index. Instead, we need to add the sizes of the fields one by one.

Great catch! I'll fix this. 🙆‍♂️